File: gitlfs-common/src/main/java/ru/bozaro/gitlfs/common/data/DeleteLockReq.java
package ru.bozaro.gitlfs.common.data;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.annotation.CheckForNull;

public final class DeleteLockReq {

  /**
   * Optional boolean specifying that the user is deleting another user's lock.
   */
  @JsonProperty(value = "force")
  @CheckForNull
  private final Boolean force;

  /**
   * Optional object describing the server ref that the locks belong to.
   */
  @JsonProperty(value = "ref")
  @CheckForNull
  private final Ref ref;

  public DeleteLockReq(
      @JsonProperty(value = "force") @CheckForNull Boolean force,
      @JsonProperty(value = "ref") @CheckForNull Ref ref) {
    this.force = force;
    this.ref = ref;
  }

  public boolean isForce() {
    return force != null && force;
  }

  @CheckForNull
  public Ref getRef() {
    return ref;
  }
}

File: gitlfs-common/src/main/kotlin/ru/bozaro/gitlfs/common/data/DeleteLockReq.kt
package ru.bozaro.gitlfs.common.data

import com.fasterxml.jackson.annotation.JsonProperty

class DeleteLockReq(
        /**
         * Optional boolean specifying that the user is deleting another user's lock.
         */
        @field:JsonProperty(value = "force") @param:JsonProperty(value = "force") private val force: Boolean?,
        /**
         * Optional object describing the server ref that the locks belong to.
         */
        @field:JsonProperty(value = "ref") @param:JsonProperty(value = "ref") val ref: Ref?) {
    fun isForce(): Boolean {
        return force != null && force
    }
}
